fastddsspy_participants/src/cpp/visualization/ModelParser.cpp

```diff
@@ -214,65 +214,219 @@ ComplexParticipantData ModelParser::participants(const SpyModel& model, const dd
 {
     std::vector<SimpleEndpointData> result;
-    // TODO
+
+    for (const auto& it : model.endpoint_database_)
+    {
+        if (it.second.is_writer())
+        {
+            result.push_back({
+                it.second.guid,
+                it.second.discoverer_participant_id ,
+                {
+                    it.second.topic.m_topic_name,
+                    it.second.topic.type_name
+                }
+            });
+        }
+    }
+
     return result;
 }
 
 std::vector<ComplexEndpointData> ModelParser::writers_verbose(const SpyModel& model)
 {
     std::vector<ComplexEndpointData> result;
-    // TODO
+
+    for (const auto& it : model.endpoint_database_)
+    {
+        result.push_back(writers(model, it.second.guid));
+    }
+
     return result;
 }
 
 ComplexEndpointData ModelParser::writers(const SpyModel& model, const ddspipe::core::types::Guid& guid)
 {
     ComplexEndpointData result;
-    // TODO
+
+    for (const auto& it : model.endpoint_database_)
+    {
+        if (it.first == guid && it.second.is_writer())
+        {
+            result = {
+                it.second.guid,
+                it.second.discoverer_participant_id ,
+                {
+                    it.second.topic.m_topic_name,
+                    it.second.topic.type_name
+                },
+                {
+                    std::to_string(it.second.topic.topic_qos.durability_qos),
+                    std::to_string(it.second.topic.topic_qos.reliability_qos)
+                }
+            };
+            break;
+        }
+    }
+
     return result;
 }
 
 
 std::vector<SimpleEndpointData> ModelParser::readers(const SpyModel& model)
 {
     std::vector<SimpleEndpointData> result;
-    // TODO
+
+    for (const auto& it : model.endpoint_database_)
+    {
+        if (it.second.is_reader())
+        {
+            result.push_back({
+                it.second.guid,
+                it.second.discoverer_participant_id ,
+                {
+                    it.second.topic.m_topic_name,
+                    it.second.topic.type_name
+                }
+            });
+        }
+    }
+
     return result;
 }
 
 std::vector<ComplexEndpointData> ModelParser::readers_verbose(const SpyModel& model)
 {
     std::vector<ComplexEndpointData> result;
-    // TODO
+
+    for (const auto& it : model.endpoint_database_)
+    {
+        if (it.second.is_reader())
+        {
+            result.push_back(readers(model, it.second.guid));
+        }
+    }
+
     return result;
 }
 
 ComplexEndpointData ModelParser::readers(const SpyModel& model, const ddspipe::core::types::Guid& guid)
 {
     ComplexEndpointData result;
-    // TODO
+
+    for (const auto& it : model.endpoint_database_)
+    {
+        if (it.first == guid && it.second.is_reader())
+        {
+            result = {
+                it.second.guid,
+                it.second.discoverer_participant_id ,
```

Comment:
Again, name incorrect. Probably join finding this value in a single function and reuse it